models/spring-ai-ollama/src/main/java/org/springframework/ai/ollama/api/OllamaApi.java

```diff
@@ -954,13 +957,15 @@ public record ProgressResponse(
 	 * Download a model from the Ollama library. Cancelled pulls are resumed from where they left off,
 	 * and multiple calls will share the same download progress.
 	 */
-	public ProgressResponse pullModel(PullModelRequest pullModelRequest) {
-		return this.restClient.post()
+	public Flux<ProgressResponse> pullModel(PullModelRequest pullModelRequest) {
```

Comment:
Using the streaming option is grew because it gives us continuous status updates on the download, that we can log to keep the user up-to-date. It also makes it easy to define timeouts and retries.